### PR TITLE
Fixes #29511 - do not run installer for f-m install/update

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

File: definitions/scenarios/packages.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 module ForemanMaintain::Scenarios
   module Packages
+    def self.skip_installer_run?(packages_list)
+      return false unless packages_list.any? { |p| p.include?('foreman_maintain') }
+      return true if packages_list.length == 1
+
+      fm_pkg = ForemanMaintain.main_package_name
+      puts "ERROR: Perform this action for '#{fm_pkg}' before any other packages."
```
Comment (upadhyeammit) [inline definitions/scenarios/packages.rb:8]: Should we change error message something like:
"ERROR: Install or Update '#{fm_pkg}' individually."

Comment (kgaikwad) [reply definitions/scenarios/packages.rb:8]: Surely! Big +1

Comment (kgaikwad) [reply definitions/scenarios/packages.rb]: Done!

**Comment by kgaikwad** [comment]
@jameerpathan111, request you to verify these changes.

**Comment by jameerpathan111** [comment]
@kgaikwad getting below error while testing this PR:

```
# foreman-maintain packages update rubygem-foreman_maintain
undefined method `any?' for "rubygem-foreman_maintain":String
```

`/var/log/foreman-maintain/foreman-maintain.log` contains:
```
D, [2020-05-11 07:24:36-0400 #16986] DEBUG -- : Running command systemctl is-enabled pulpcore-api 2>&1 | tail -1 with stdin nil
D, [2020-05-11 07:24:36-0400 #16986] DEBUG -- : output of the command:
 Failed to get unit file state for pulpcore-api.service: No such file or directory
I, [2020-05-11 07:24:36-0400 #16986]  INFO -- : Running foreman-maintain command with arguments [["packages", "update", "foreman-maintain"]]
E, [2020-05-11 07:24:36-0400 #16986] ERROR -- : undefined method `any?' for "foreman-maintain":String (NoMethodError)
/root/foreman_maintain/definitions/scenarios/packages.rb:4:in `skip_installer_run?'
/root/foreman_maintain/definitions/scenarios/packages.rb:87:in `compose'
/root/foreman_maintain/lib/foreman_maintain/scenario.rb:86:in `initialize'
/root/foreman_maintain/lib/foreman_maintain/cli/packages_command.rb:45:in `new'
/root/foreman_maintain/lib/foreman_maintain/cli/packages_command.rb:45:in `execute'
/usr/share/gems/gems/clamp-0.6.2/lib/clamp/command.rb:67:in `run'
/usr/share/gems/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/share/gems/gems/clamp-0.6.2/lib/clamp/command.rb:67:in `run'
/usr/share/gems/gems/clamp-0.6.2/lib/clamp/subcommand/execution.rb:11:in `execute'
/usr/share/gems/gems/clamp-0.6.2/lib/clamp/command.rb:67:in `run'
/root/foreman_maintain/lib/foreman_maintain/cli.rb:34:in `run'
/usr/share/gems/gems/clamp-0.6.2/lib/clamp/command.rb:125:in `run'
/root/foreman_maintain/bin/foreman-maintain:15:in `<main>'
I, [2020-05-11 07:24:36-0400 #16986]  INFO -- : foreman-maintain command finished with 1

```

**Comment by kgaikwad** [comment]
Thanks @jameerpathan111 for testing. Fixed the error.

**Comment by jameerpathan111** [comment]
@kgaikwad I have tried this PR and it works as expected.

Steps I performed:
- downgrade foreman-maintain version
- run `foreman-maintain packages install rubygem-foreman_maintain`.
- downgrade foreman-maintain version
- run `foreman-maintain packages update rubygem-foreman_maintain`.

Observation:
- foreman-maintain package got updated and satellite-installer was not executed after package update.
- Tried to install/update one more package along with `rubygem-foreman_maintain`, it only updated `rubygem-foreman_maintain` and didn't install other package.
- Not related to this PR: While updating/installing package using `foreman-maintain packages update` it always runs `yum clean all` command, where as `foreman-maintain packages install` doesn't.

**Pull request merged**